addons/point_of_sale/models/pos_session.py

```diff
@@ -1319,23 +1323,38 @@ def _get_invoice_receivable_vals(self, amount, amount_converted):
             'account_id': self.company_id.account_default_pos_receivable_account_id.id,
             'move_id': self.move_id.id,
             'name': _('From invoice payments'),
+            'display_type': 'payment_term',
         }
         return self._credit_amounts(partial_vals, amount, amount_converted)
 
-    def _get_sale_vals(self, key, amount, amount_converted):
-        account_id, sign, tax_ids, base_tag_ids = key
+    def _get_sale_vals(self, key, sale_vals):
+        account_id, sign, tax_ids, base_tag_ids, product_id = key
+        amount = sale_vals['amount']
+        amount_converted = sale_vals['amount_converted']
         applied_taxes = self.env['account.tax'].browse(tax_ids)
+        if product_id:
```

Comment:
Hm product is required on pos order line

Reply:
it's based on configuration.
`self.config_id.is_closing_entry_by_product`
if this is true then product_id is there in key.